app/src/main/java/com/alexP/socialnetwork/ui/screens/base/BaseFragment.kt

```diff
@@ -0,0 +1,36 @@
+package com.alexP.socialnetwork.ui.screens.base
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.activity.enableEdgeToEdge
+import androidx.fragment.app.Fragment
+import androidx.viewbinding.ViewBinding
+import com.alexP.socialnetwork.utils.applyWindowInsets
+
+
+abstract class BaseFragment<out VB : ViewBinding> : Fragment() {
+
+    private var _binding: VB? = null
+    val binding
+        get() = _binding!!
+
+    abstract fun inflate(inflater: LayoutInflater, container: ViewGroup?): VB
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        requireActivity().enableEdgeToEdge()
```

Comment:
It is better to remove [enableEdgeToEdge](https://developer.android.com/develop/ui/views/layout/edge-to-edge#enable-edge-to-edge-display)() function from here and paste in MainActivity onCreate() function. So you will have one activity with enabled edge to edge instead of many fragments with enabled edgetoedge.